e2e/specs/websocket_reconnects.spec.js

```diff
@@ -44,26 +44,51 @@ describe("websocket reconnects", () => {
         }, 100);
       });
 
-      // Wait until we've disconnected.
+      // Wait until we've disconnected and reconnected.
       cy.get("[data-testid='stStatusWidget']").should(
         "have.text",
-        "Connecting"
+        "Connecting",
       );
-      // Wait until we've reconnected and rerun the script.
       cy.get("[data-testid='stStatusWidget']").should("not.exist");
 
-      cy.get(".stMarkdown").contains(`count: ${expectedCount}`);
+      cy.getIndexed(".stMarkdown", 0).contains(`count: ${expectedCount}`);
+      cy.getIndexed(".stMarkdown", 1).contains(`slow operations attempted: 0`);
     }
   });
 
+  it("reruns script when script is interrupted by a websocket disconnect", () => {
+    cy.get(".stCheckbox input").click({ force: true });
+
+    cy.window().then((win) => {
+      setTimeout(() => {
+        win.streamlitDebug.disconnectWebsocket();
+      }, 100);
+    });
+
+    // Wait until we've disconnected and reconnected.
+    cy.get("[data-testid='stStatusWidget']").should("have.text", "Connecting");
+    cy.get("[data-testid='stStatusWidget']").should(
+      "not.have.text",
+      "Connecting",
+    );
+
+    // Counter was incremented once when the checkbox was selected and a second
+    // time on the script rerun. We set a high timeout for this check because
+    // the operation we're simulating is naturally slow to give us time to
+    // disconnect the websocket while it's running.
+    cy.getIndexed(".stMarkdown", 1).contains(`slow operations attempted: 2`, {
+      timeout: 10000,
+    });
+  });
+
   it("retains uploaded files when the websocket connection is dropped and reconnects", () => {
     const fileName1 = "file1.txt";
     const uploaderIndex = 0;
 
     cy.fixture(fileName1).then((file1) => {
       cy.getIndexed(
         "[data-testid='stFileUploaderDropzone']",
-        uploaderIndex
+        uploaderIndex,
```

Comment:
Is this file ignored by prettier? And if it is, should it be?

Reply:
Hmm, just looked into this, and it looks like prettier does only run on files in the `frontend/` directory, so this file does end up being ignored when prettier runs in CI, but my autoformatter ran when I changed this file.

I think this is probably fine to not worry about fixing at this point, though, given that we're hopefully going to finish migrating off cypress in the next quarter or so anyway. (I'll circle back around to migrating this test entirely if I happen to have time in the next few weeks since we're wrapping up the quarter)